aiohttp/client_reqrep.py

```diff
@@ -412,12 +414,11 @@ def update_auto_headers(self, skip_auto_headers: Optional[Iterable[str]]) -> Non
         else:
             # Fast path when there are no headers to skip
             # which is the most common case.
-            self.skip_auto_headers = CIMultiDict()
             used_headers = self.headers
 
         for hdr, val in self.DEFAULT_HEADERS.items():
             if hdr not in used_headers:
-                self.headers.add(hdr, val)
+                self.headers[hdr] = val
```

Comment:
We are already checking if the header exists in `used_headers` which is the combination of `self.headers` and `skip_auto_headers` so we can use `[]` instead of `.add()` which is a bit faster